src/modules/cmdpal/Exts/MicrosoftLearnExtension/Pages/MicrosoftLearnExtensionPage.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.CmdPal.Extensions;
+using Microsoft.CmdPal.Extensions.Helpers;
+using Windows.UI.WebUI;
+
+namespace MicrosoftLearnExtension;
+
+internal sealed partial class MicrosoftLearnExtensionPage : DynamicListPage
+{
+    private static readonly string EndPointUrl = "https://learn.microsoft.com/api/search";
+
+    // TODO: Provide options to search different locales and products
+    private static readonly string Locale = "en-us";
+    private static readonly string Product = "/devrel/b3627ad5-057f-488f-8b97-08be347b4be5"; // Windows App SDK
+
+    private static readonly JsonSerializerOptions JsonOptions = new()
+    {
+        PropertyNameCaseInsensitive = true,
+    };
+
+    public MicrosoftLearnExtensionPage()
+    {
+        Icon = new("https://learn.microsoft.com/favicon.ico");
+        Name = "Microsoft Learn Doc Search";
+        PlaceholderText = "Search Windows App SDK docs";
+        IsLoading = true;
+
+        // #091f2c
+        AccentColor = ColorHelpers.FromRgb(9, 31, 44);
+    }
+
+    private static async Task<List<SearchResult>> SearchMicrosoftLearn(string query)
+    {
+        if (string.IsNullOrWhiteSpace(query))
+        {
+            return new();
+        }
+
+        //// TODO: Sanitize input?
+        using HttpClient client = new HttpClient();
+
+        var response = await client.GetStringAsync($"{EndPointUrl}?locale={Locale}&$filter=products/any(product:product eq '{Product}')&search={query}");
+
+        return JsonSerializer.Deserialize<SearchResultList>(response, JsonOptions).Results.ToList();
+    }
+
+    public override IListItem[] GetItems()
+    {
+        var t = DoGetItems(SearchText);
+        t.ConfigureAwait(false);
+        return t.Result;
+    }
+
+    public override void UpdateSearchText(string oldSearch, string newSearch)
+    {
+        IsLoading = true;
+        GetItems();
```

Comment:
IMO I would do the search here, build the results, cache them into a member, then `RaiseItemsChanged`. Then in `GetItems()`, just return the cached member

Reply:
Alright that works better, but I'm also trying to list some portals at the start now, but they show up more often than I would expect. Like in when the query is being fetched and when backspacing instead of re-triggering query.

I feel like I'm being dense here...